faunadb-scala/src/main/scala/faunadb/values/time/package.scala

```diff
@@ -0,0 +1,89 @@
+package faunadb.values
+
+import org.joda.time.Instant
+import org.joda.time.format.ISODateTimeFormat
+import scala.Ordered._
+
+package time {
+
+  /**
+    * Represents an instance of [[org.joda.time.Instant]] with nanosecond precision.
+    * It calculates the overflow from nano to milliseconds and adds to the initial instant.
+    */
+  class HighPrecisionTime private(initialInstant: Instant, nanosToAdd: Long) {
```

Comment:
We could just copy the internal storage of java 8 instants: seconds since epoch (can be negative), plus nanoseconds. The java 8 instant constructor adjusts the seconds/nanoseconds in order to keep the range of nanoseconds to within 0 and 999_999_999.